matplotlib-chord.py

```diff
@@ -162,13 +164,14 @@ def chordDiagram(X, ax, colors=None, width=0.1, pad=2, chordwidth=0.7):
     x = X.sum(axis = 1) # sum over rows
     ax.set_xlim(-1.1, 1.1)
     ax.set_ylim(-1.1, 1.1)
-
-    if colors is None:
-    # use d3.js category10 https://github.com/d3/d3-3.x-api-reference/blob/master/Ordinal-Scales.md#category10
+    
+    import matplotlib.pyplot as plt
+    if (len(x) > 10) or (colors is None):
+        cm = plt.get_cmap(cmap) 
+        colors = cm(np.linspace(0,1,len(x)))[:,:3]
+    else:
```

Comment:
This "else" will apply only when `len(x)<=10 and (colors is not None)`.  The `colors` will be changed in the following codes, hence this seems to be a bug.

My suggestion:
```python
if colors is None:
    if (len(x) > 10) or (cmap is not None):
        cm = plt.get_cmap(cmap)
        colors = cm(np.linspace(0,1,len(x)))[:,:3]
    else:
        # use default d3.js category10
```

Reply:
In this case, it is when `if (len(x) > 10) or (colors is None):`, implying **or**, not and. I mean, if any of these two conditions are met, we use the default colormap. I don't know if this is what you meant. Your suggestion would have issues when len(x)>10 and len(colors)<=10, although we could work out a solution.

Reply:
In my original version, I assume that users make sure that `len(colors)=len(x)` , if they use `colors`. Therefore, I didn't expect the issue `len(x)>10` and `len(colors)<=10`.

My original idea is that:
(1) if `colors` is None
     (1a) if `len(x)<=10`, use d3.js category10 colors;
     (1b) else, quit. I used to think that it is not a good idea to plot a chord diagram with too many categories, which makes the chords too many to see individual ones. Maybe I am wrong at this point. At lease, I have the keyword `colors` to rescue this.
(2) if `colors` is given by users, `len(colors)=len(x)`, and the colors are sometimes meaningful, e.g., colors are chosen to be consistent with other panels. In this case, everything is left to the users.

We can think about the following options:
1. Maybe we can use your cmap if len(x)>10, and `d3.js category10` if len(x)<10?
2. Maybe we can use `d3.js category20` colors if 10<len(x)<=20; and raise an error if len(x)>20?

Let's also `assert len(colors)==len(x)` if `not colors is None`.

What do you think?

Reply:
Yes, I think the best idea is prioritize "colors" vs "colormap". I mean, by default (with no colormap or colors specified) use category10 if `len(x)<=10`, category20 if `10<len(x)<=20` and `gist_rainbow` (better than rainbow) if `len(x)>20`. It is indeed difficult to visualize if there are more than 15 categories, but sometimes it is inevitable. 

I guess I will remove the colormap option and use colors instead. I mean:
```Python
if type(colors) is list:
      # use the list of colors (and `assert len(colors)==len(x)`)
elif type(colors) is str:
      # use the colormap specified
else: 
      #use the default colors specified
```
It should be easier that way. What do you think about this solution?

Reply:
Great idea. I think this solution will also simplify the user interface for the function. Please implement it. Thanks a lot.